src/schema/v1.json

```diff
@@ -11,8 +11,6 @@
    "coreos-assembler.code-source",
    "coreos-assembler.config-dirty",
    "coreos-assembler.config-gitrev",
-   "coreos-assembler.container-config-git",
-   "coreos-assembler.container-image-git",
```

Comment:
Hmm, I think this needs to go in `optional`? Otherwise, once we turn on `additionalProperties: false` in #1084, builds will start erroring out.

Kinda feels like we should distinguish between the "stable" parts of the schema, and the extra bits. E.g. something like, anything starting with `coreos-assembler.*` or `fedora-coreos.*` or `rhcos.*` could just be allowed (though I wouldn't be surprised if some of those keys are already used in the wild too). This is partly why with FCOS, it transforms the `meta.json` into a more narrow schema, so we don't have to make that declaration and only expose what we need to.

Reply:
Question: what denotes stable and not?  These values were used in both the RHCOS and the FCOS build systems. I would argue they are stable. 

Part of the point in enforcing the schema was because we have undefined things relying on parts of the schema that break stuff. 

Can you please move them to optional? I'll do some more work on this to allow for easier schemas.

Reply:
Ah, yep thanks - moved to optional.

Reply:
The commit message describes the scenario in which these keys won't exist, right?

I think we're all agreed on moving them to optional, so that's done below :arrow_down:

Reply:
Yeah, patch here makes sense!

> Question: what denotes stable and not? These values were used in both the RHCOS and the FCOS build systems. I would argue they are stable.

Basically whether the keys are intended for us, or for consumption. E.g., anything in [`rhcos.json`](https://github.com/openshift/installer/blob/master/data/data/rhcos.json) is clearly being consumed. And I'm sure there are a few other keys as well that end up elsewhere. But everything else is basically for our use.

I think the key difference (pun intended :) ) is that metadata for us is just used by code under our direct control. It doesn't technically need to be stable. If e.g. the way cosa works in the future changes and `coreos-assembler.image-input-checksum` no longer makes sense, that's purely an implementation detail and we're free to drop that. OTOH, we need to be much more strict about changing e.g. `amis`.

I mean, in the case of RHCOS, it's not a bright line either, since we have more control over the whole stack than in FCOS. But still, there's some degrees there...

The tricky bit is how that gets expressed in the schema. The glob suggestion is just one idea.